assignments/array-methods.js

```diff
@@ -55,29 +55,43 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
-let fullName = [];
-console.log(fullName);
+let fullNameArray = [];
+let fullName = runners.forEach(name => fullNameArray.push(`${name.first_name} ${name.last_name}`));
+
+console.log(fullNameArray);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
-console.log(allCaps); 
+let allCaps = runners.map(name => name.first_name.toUpperCase());
+
+console.log(allCaps);
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+let largeShirts = runners.filter(largeShirt => largeShirt.shirt_size = 'L');
+
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+let ticketPriceTotal = runners.reduce((priceTotal, items) => priceTotal + items.donation, 0);
+
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
-// Problem 1
+// Problem 1. The run has been rescheduled due to weather conditions that are unfit to be out in, get a list of the first name and email of each runner.
+let firstEmail = runners.map(item => `${item.first_name} ${item.email}`);
+
+console.log(firstEmail);
+
+// Problem 2 The event director needs a list of all the company names to see which companies are participating in the run.
+let company = runners.map(item => `${item.company_name}`);
```

Comment:
This is interesting. But this array will result in having duplicates. It will be an interesting exercise to get rid of the duplicates.